Task6/posts.js

```diff
@@ -0,0 +1,368 @@
+                for (let i = skip; i < copy.length; i++) {
+                    if (copy[i].author === filterConfig.author) {
+                        for (let j = 0; j < filterConfig.hashTags.length; j++) {
+                            if (copy[i].hashTags.indexOf(filterConfig.hashTags[j]) >= 0) {
+                                temp++;
+                            }
+                        }
+                        if (temp === filterConfig.hashTags.length) {
+                            searching.push(copy[i]);
+                            count++;
+                        }
+                        temp = 0;
+                        if (count === top) {
+                            break;
+                        }
+                    }
+                }
+                if (searching.length === 0) {
+                    return 'NO SUCH ELEMENTS';
+                }
+            } else {
+                return 'Incorrect data';
+            }
+        }
+        return searching;
+    }
+
+    function getPost(id) {
+        if (!isInteger(parseInt(id, 10))) {
+            return 'Incorrect data';
+        }
+        if (isInteger(parseInt(id, 10)) && id > 0) {
+            for (var i = 0; i < posts.length; i++) {
+                if (parseInt(id, 10) === parseInt(posts[i].id, 10)) {
+                    return posts[i];
+                }
+            }
+            return new Object('NO POST');
+        } else {
+            return 'Incorrect data';
+        }
+    }
+
+    function validatePost(post) {
+        if (typeof post != 'object') {
+            return 'Incorrect data';
+        } else if (typeof post.description != "string") {
+            return false;
+        } else if (toString.call(post.createdAt) != "[object Date]") {
+            return false;
+        } else if (typeof post.author != "string") {
+            return false;
+        } else if (toString.call(post.hashTags) != "[object Array]") {
+            return false;
+        } else if (toString.call(post.likes) != "[object Array]") {
+            return false;
+        } else return true;
+    }
+
+    function addPost(post) {
+        if (typeof post != 'object') {
+            return 'Incorrect data';
+        }
+        if (!validatePost(post))
+            return false;
+        posts.push(post);
+        return getPost(post.id) != 'NO POST';
+    }
+
+    function editPost(id, post) {
+        if (typeof post != 'object' || !isInteger(parseInt(id, 10))) {
+            return 'Incorrect data';
+        }
+        let postToEdit = getPost(parseInt(id, 10) - 1);
+        if (post.hasOwnProperty('description')) {
+            postToEdit.description = post.description;
+        }
+        if (post.hasOwnProperty('hashTags')) {
+            postToEdit.hashTags = post.hashTags;
+        }
+        if (post.hasOwnProperty('photoLink')) {
+            postToEdit.photoLink = post.photoLink;
+        }
+        return validatePost(postToEdit);
```

Comment:
You've already changed the post, but after that you are validating it. So, if validation fails, post was already changed